docs/userguide/failover/cluster-taint-management.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Cluster Taint Management
+---
+
+Taints in a Cluster, specifically configured in the `.spec.taints` field of the `Cluster` object, have a significant impact on application scheduling and operation. For instance, if a cluster has a `NoSchedule` taint, it can prevent new applications from being scheduled onto that cluster. The effect of a taint on an application depends on the `Effect` configuration of the taint.
+
+Karmada supports different types of taints based on their `Effect` values:
+- `NoSchedule`: Prevents new applications from scheduling to the cluster unless the application explicitly tolerates it.
+- `NoExecute`: Prevents new scheduling **and evicts existing applications** from the cluster unless the application explicitly tolerates it.
+
+Note that the eviction feature of NoExecute is disabled by default, to enable it, you need to use the `--enable-no-execute-taint-eviction=true` flag in the `karmada-controller-manager` and also explicitly enable the `Failover` feature gate.
+
+This document explains how Karmada manages taints on Clusters, including
+- Customized taint management based on cluster conditions.
+- Taints managed by the Karmada system.
+- Configuring application tolerations to bypass taint effects.
+
+## Customized Taint Management
+
+The `ClusterTaintPolicy` API allows administrators to define rules for applying or removing taints from clusters based on specific conditions. This is particularly useful for implementing advanced schedul ing policies or failover strategies.
+
+Cluster conditions can be determined not only by the Karmada system but also by any third-party system. This offers administrators great flexibility in defining cluster health criteria. For example, the Karmada system set the `Ready` condition based on its built-in health checks. However, a third-party system can introduce additional conditions,
+
+such as `SchedulerDown`, to indicate that the cluster scheduler is non-functional. When combined with `ClusterTaintPolicy`, these custom conditions can be used to prevent new applications from being scheduled onto affected clusters, providing more granular control over application placement and enhancing overall system reliability.
```

Comment:
```suggestion
Cluster conditions can be determined not only by the Karmada system but also by any third-party system. This offers administrators great flexibility in defining cluster health criteria. For example, the Karmada system sets the `Ready` condition based on its built-in health checks. However, a third-party system can introduce additional conditions, such as `SchedulerDown`, to indicate that the cluster scheduler is non-functional. When combined with `ClusterTaintPolicy`, these custom conditions can be used to prevent new applications from being scheduled onto affected clusters, providing more granular control over application placement and enhancing overall system reliability.
```